js.js

```diff
@@ -26,3 +26,151 @@ list.forEach((e) => {
+      <p class="work-p">${project.description}</p>
+      <ul class="blue-list">
+        <li class="blue-list li">${project.technologies[0]}</li>
+        <li class="blue-list li">${project.technologies[1]}</li>
+        <li class="blue-list li">${project.technologies[2]}</li>
+      </ul>
+      <button class="work-btn btn">see project</button>
+    </div>
+  `;
+
+  existingCard.appendChild(card);
+});
+
+// popup section
+projects.forEach((project) => {
+  const popup = document.createElement('div');
+  popup.classList.add('modal');
+  popup.innerHTML = `
+    <div class="modal-close">
+      <h1 class="modal-h1">${project.name}</h1>
+      <i class="fa-solid fa-xmark"></i>
+    </div>
+    <ul class="modal-first-list">
+      <li class="list1">${project.list[0]}</li>
+      <li class="list2">${project.list[1]}</li>
+      <li class="list2">${project.list[2]}</li>
+    </ul>
+    <img class="modal-img" src="${project.image}" alt="portfolio img">
+    <div class="modal-text">
+    <p class="modal-para">${project.popuptext}</p>
+    <p class="modal-para2">${project.popuptext2}</p>
+     <div class="modal-list-btn">
+    <ul class="modal-blue-list">
+      <li class="modal-blue-list li">${project.technologies[0]}</li>
+      <li class="modal-blue-list li">${project.technologies[1]}</li>
+      <li class="modal-blue-list li"> ${project.technologies[2]}</li>
+      <li class="modal-blue-list li modal-blue-list2">${project.technologies2[0]}</li>
+      <li class="modal-blue-list li modal-blue-list2">${project.technologies2[1]}</li>
+    </ul>
+    <div class="modal-btn-flex">
+      <a  href="${project.liveLink}><button " class="btn modal-btn"> see live <i class="fa-solid fa-arrow-up-right-from-square"></i></button></a>
+      <button href="${project.sourceLink}" class="btn modal-btn"> see source <i class="fa-brands fa-github"></i></button>
+    </div>
+    </div>
+    </div>
+  `;
+  existingCard.appendChild(popup);
+});
+// Modal popups
+const btns = document.querySelectorAll('.work-btn');
+const modals = document.querySelectorAll('.modal');
+
+btns.forEach((btn, index) => {
+  btn.addEventListener('click', () => {
+    modals[index].classList.toggle('active');
+  });
+});
+
+modals.forEach((modal) => {
+  const closeButton = modal.querySelector('.fa-solid');
+  closeButton.addEventListener('click', () => {
+    modal.classList.toggle('active');
+  });
+});
+
+// the form validition
+const form = document.querySelector('#form');
+const emailRegex = /^[^@\s]+@[^@\s]+\.[a-z]+$/i;
+const errorMessage = document.querySelector('#error');
+const errorclose = document.querySelector('.error-close');
+// the event status
+form.addEventListener('submit', (event) => {
+  const inputEmailValue = document.querySelector('#email').value;
+  if (!emailRegex.test(inputEmailValue)) {
+    event.preventDefault();
+    errorMessage.style.display = 'block';
+  }
+
+  errorclose.addEventListener('click', () => {
+    errorMessage.style.display = 'none';
+  });
+});
```

Comment:
- [ ] The contact form should have at least one validation checking if the email is written in all lowercase. That means that If a user submits the contact form with the email field containing **uppercase** characters an error message should show up. But If a user submits the contact form with the email field containing only **lowercase** characters the form is sent normally. Currently, I can submit the form even when the email field contains **uppercase** characters. Kindly fix this. 

❗ **You can check if the email is all lowercase by comparing if the email and its lowercase converted version are the same, ie: `email === email.toLowerCase()`, utilizing regular expressions, or looping through each letter of the email string. Any implementation of this check is considered valid as long as it works properly.**